### PR TITLE
validation message for school search field

### PR DESCRIPTION
**Pull request by hughes-dustin**
## Description
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/2381

## Testing done
Developer locally tested.

## Screenshots
Screen Shot 2019-10-15 at 10 38 19 AM

## Acceptance criteria
- [ ] As a screen reader user, I want to hear an error message read out when I press `Search Schools` without entering a term like "University of Mississippi" in the search field.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]

File: src/applications/gi/containers/LandingPage.jsx [hunk 1]
```diff
@@ -37,9 +43,19 @@ export class LandingPage extends React.Component {
     // Only search upon blur, keyUp, suggestion selection
```
Comment (shawkey-daniel) [inline src/applications/gi/containers/LandingPage.jsx]: field is not used

File: src/applications/gi/containers/LandingPage.jsx [hunk 2]
```diff
@@ -37,9 +43,19 @@ export class LandingPage extends React.Component {
...
+      this.setState({
+        searchError: false,
+      });
       this.search(value);
+    } else {
+      this.setState({
+        searchError: true,
+      });
     }
   };
```
Comment (shawkey-daniel) [inline src/applications/gi/containers/LandingPage.jsx]: this.setState({
      searchError: !(isVetTecSelected(this.props.filters) || value),
    });

File: src/applications/gi/containers/LandingPage.jsx [hunk 3]
```diff
@@ -111,6 +127,18 @@ export class LandingPage extends React.Component {
...
+    if (searchQuery === '') {
+      this.setState({
+        searchError: true,
+      });
+    } else {
+      this.setState({
+        searchError: false,
+      });
+    }
+  };
```
Comment (shawkey-daniel) [inline src/applications/gi/containers/LandingPage.jsx]: this.setState({
      searchError: searchQuery !== '',
    });

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]
Can we use the platform alias on line 4 of KeywordSearch.jsx please?

File: src/applications/gi/containers/LandingPage.jsx (hunk 1 above)
Comment (hughes-dustin) [reply src/applications/gi/containers/LandingPage.jsx]: Removed useless field.

File: src/applications/gi/containers/LandingPage.jsx (hunk 2 above)
Comment (hughes-dustin) [reply src/applications/gi/containers/LandingPage.jsx]: Implemented.

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]

File: src/applications/gi/containers/LandingPage.jsx [hunk 4]
```diff
@@ -111,6 +118,12 @@ export class LandingPage extends React.Component {
     this.props.eligibilityChange(e);
   };
 
+  validateSearchQuery = searchQuery => {
+    this.setState({
+      searchError: searchQuery === '',
+    });
```
Comment (U-DON) [inline src/applications/gi/containers/LandingPage.jsx]: Should `LandingPage` keep track of this state or would it be better to be delegated to `KeywordSearch`, where you wouldn't have to be concerned with passing the props anywhere else the component is used?

Also, it seems like the `setState` will get called twice for any changes, and the error state is checked differently between the two calls in `KeywordSearch`:

1. `handleKeyUp`, setting `searchError: !(isVetTecSelected(this.props.filters) || value)`
2. `handleChange`, setting `searchError: searchQuery === ''`.

Not sure if that will be desirable.

**Review by U-DON — CHANGES_REQUESTED** [review]

File: src/applications/gi/containers/LandingPage.jsx (hunk 4 above)
Comment (hughes-dustin) [reply src/applications/gi/containers/LandingPage.jsx]: I know this seems quirky and believe me I tried to encompass the validation logic all in one component. But I couldn't figure out how to do it without breaking functionality or creating another 508 defect.

There were a number of issues that kept coming up where the validation message would persist even after data was inputted in the input field. Or if you stepped back to the `Landing Page` from `Search Results` the validation message error would immediately appear. 

Then there was the user case where if you select `Vet Tec` on the `Landing Page` which shouldn't display a validation error message on `Form Submit` when the Search Field is empty.

**Review by U-DON — APPROVED** [review]

**Pull request merged**